src/requirements/reqs-functions.ts

```diff
@@ -4,43 +4,14 @@ import {
   BaseRequirement,
   DecoratedCollegeOrMajorRequirement,
   RequirementFulfillment,
+  RequirementFulfillmentStatistics,
   GroupedRequirementFulfillmentReport
 } from './types';
 
 type RequirementMap = { readonly [code: string]: readonly string[] };
 type MutableRequirementMap = { [code: string]: readonly string[] };
 type MutableRequirementMapWithMutableChildren = { [code: string]: string[] };
 
-/**
- * @param requirement : the requirement information as object
- * @param totalRequirementCredits : total credits of courses that satisfied requirement
- * @param totalRequirementCount : total number of courses that satisfied requirement
- * @param coursesThatFulilledRequirement : courses that satisfied requirement
- * @returns {RequirementFulfillment}
- */
-function createRequirementJSON(
-  requirement: BaseRequirement,
-  totalRequirementCredits: number,
-  totalRequirementCount: number,
-  coursesThatFulilledRequirement: readonly CourseTaken[][]
-): RequirementFulfillment {
-  let fulfilled: number | undefined;
-  switch (requirement.fulfilledBy) {
-    case 'courses':
-      fulfilled = totalRequirementCount;
-      break;
-    case 'credits':
-      fulfilled = totalRequirementCredits;
-      break;
-    case 'self-check':
-      fulfilled = undefined;
-      break;
-    default:
-      throw new Error('Fulfillment type unknown.');
-  }
-  return { requirement, courses: coursesThatFulilledRequirement, fulfilled };
-}
-
```

Comment:
Definitely makes the code much cleaner. Requirements JSON like this was definitely not necessary in the equation